enhancements/manifestlist/manifestlist-support.md

```diff
@@ -0,0 +1,277 @@
+information.
+
+#### Image registry pull through authorization
+
+The current Image and ImageStream layouts are leveraged by the image registry
+when verifying if one is allowed or not to execute a pull through for a certain
+blob. APIs needed to be adjusted to take into account this extra level of
+indirection.
+
+Before providing the client with content image registry goes through a series
+of checks to verify if a valid reference for the blob (by sha) exists and is
+valid within the namespace (repository). Image registry currently caches this
+information but some changes are going to be needed to validate using the newly
+introduced "child images".
+
+## Design Details
+
+### Proposed Image API changes
+
+Images are indexed (named) in OpenShift by the hash of their content. Currently
+an Image is represented by the following structure:
+
+```go
+type Image struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	DockerImageReference string
+	DockerImageMetadata DockerImage
+	DockerImageMetadataVersion string
+	DockerImageManifest string
+	DockerImageLayers []ImageLayer
+	Signatures []ImageSignature
+	DockerImageSignatures [][]byte
+	DockerImageManifestMediaType string
+	DockerImageConfig string
+}
+```
+
+This document proposes the introduction of an extra field pointing to all
+"child images":
+
+```go
+	DockerImageManifests []ImageManifest
+```
+
+`ImageManifest` struct would look like the following:
+
+```go
+type ImageManifest struct {
+	Digest       string
+	MediaType    string
+	ManifestSize int64
+	Architecture string
+	OS           string
+	Variant      string
+}
+```
+
+The Image pointed by the `Digest` property contain all regular information
+(i.e. Config) we can see nowadays in a regular Image.
+
+Not all fields in an Image are relevant for a manifest list, therefore other
+proposed change is to add an "omitempty" flag on `DockerImageLayers` property
+as well as to make `DockerImageMetadata` a pointer.
+
+The resulting Image struct would look like:
+
+```go
+type Image struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+	DockerImageReference string
+	DockerImageMetadata *DockerImage
+	DockerImageMetadataVersion string
+	DockerImageManifest string
+	DockerImageLayers []ImageLayer
+	Signatures []ImageSignature
+	DockerImageSignatures [][]byte
+	DockerImageManifestMediaType string
+	DockerImageConfig string
+	DockerImageManifests []ImageManifest
+}
+
+```
+
+### Test Plan
+
+Other to regular unit tests we gonna need to back this feature with an end
+to end test. This test would execute the following operations:
+
+1. Deploy a temporary image registry.
+2. Publish a manifest list containing images for different platforms.
+3. Create an ImageStream object to import the published manifest list.
+4. Create a Deployment using the ImageStream.
+
+A second test would be very similar to this one but at this time leveraging
+pull through. Podman currently supports managing manifest lists and we can
+use its internals for step 2.
```

Comment:
We also need to test that a manifest list can be mirrored into the integrated registry using `oc image mirror`.

Reply:
Added.